### PR TITLE
feat: add C `ndarray` implementation for `blas/base/caxpy`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request adds C `ndarray` implementation for `caxpy`. 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/caxpy/src/caxpy_ndarray.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#include "stdlib/blas/base/caxpy.h"
+#include "stdlib/blas/base/scabs1.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/complex/float32/ctor.h"
+#include "stdlib/complex/float32/base/add.h"
+#include "stdlib/complex/float32/base/mul.h"
+#include <stdint.h>
+
+/**
+*Scales a single-precision complex floating-point vector by a single-precision complex floating-point constant and adds the result to a single-precision complex floating-point vector using alternative indexing semantics.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/caxpy/src/caxpy_ndarray.c]:
```suggestion
* Scales a single-precision complex floating-point vector by a single-precision complex floating-point constant and adds the result to a single-precision complex floating-point vector using alternative indexing semantics.
```

File: lib/node_modules/@stdlib/blas/base/caxpy/src/caxpy_ndarray.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	stdlib_complex64_t x;
+	stdlib_complex64_t y;
+	int64_t is1;
+	int64_t is2;
+	int64_t i;
+
+	if ( N <= 0 ) {
+		return;
+	}
+	if( c_scabs1( ca ) == 0.0f ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/caxpy/src/caxpy_ndarray.c]:
```suggestion
	if ( c_scabs1( ca ) == 0.0f ) {
```

**Review by kgryte — COMMENTED** [review]
Thanks for working on this, @aman-095. IIRC, we also need to add documentation of the C APIs to the README, correct?

**Comment by aman-095** [comment]
@kgryte, updated.

File: lib/node_modules/@stdlib/blas/base/caxpy/README.md [hunk 3]
```diff
@@ -229,6 +229,152 @@ logEach( '(%s)*(%s) + (%s) = %s', ca, cx, cyc, cy );
...
+Scales values from `cx` by `ca` and adds the result to `cy` using alternative indexing semantics.
+
+```c
+#include "stdlib/complex/float32/ctor.h"
+
+float cx[] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
+float cy[] = { -1.0f, -2.0f, -3.0f, -4.0f, -5.0f, -6.0f, -7.0f, -8.0f }
+const stdlib_complex64_t ca = stdlib_complex64( 2.0f, 2.0f );
+
+c_caxpy( 4, ca, (void *)cx, 1, 0, (void *)cy, 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/caxpy/README.md]:
```suggestion
c_caxpy_ndarray( 4, ca, (void *)cx, 1, 0, (void *)cy, 1, 0 );
```

File: lib/node_modules/@stdlib/blas/base/daxpy/README.md [hunk 4]
```diff
@@ -207,7 +207,7 @@ The function accepts the following arguments:
 -   **X**: `[in] double*` input array.
 -   **strideX**: `[in] CBLAS_INT` index increment for `X`.
 -   **Y**: `[inout] double*` output array.
--   **strideY**: `[in CBLAS_INT` index increment for `Y`.
+-   **strideY**: `[in] CBLAS_INT` index increment for `Y`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/daxpy/README.md:210]: Thanks for fixing this. In the future, however, we'll want to open a separate PR in order to ensure a clean commit history and changelog.

**Review by kgryte — APPROVED** [review]
Thanks for working on this, @aman-095. LGTM.

**Pull request merged**